### PR TITLE
docstring standardization

### PR DESCRIPTION
**Pull request by steenzout**
Hello,

  I've noticed the docstrings weren't following any convention so I made some changes and made them consistent.
if you want to follow a specific convention let me know and I'll go through the code again.
  There was some text on the docstrings that was broken in a way that made it difficult to read.
I've added some newlines to help with that.

  I've also added a tox env so you can build the docs using

```
$ tox -e docs -- html
```

  I added some :param on some docstrings and fixed a few complaints from sphinx so the HTML documentation should also look better.

  hope it will be useful.

cheers!

### CODE REVIEW

**Comment by erikbern** [comment]
LGTM!

**Comment by Tarrasch** [comment]
Can you rebase so we can merge?

File: tox.ini [hunk 1]
```diff
@@ -29,3 +29,17 @@ commands = pep8 --ignore E501 luigi test examples bin
 [testenv:autopep8]
 deps = autopep8
 commands = autopep8 --ignore E501 -a -i -r luigi test examples bin
+
+[testenv:docs]
+# to call this use `tox -e docs -- html`
```
Comment (Tarrasch) [inline tox.ini]: Awesome that you are explicit here :)

File: tox.ini [hunk 2]
```diff
@@ -29,3 +29,17 @@ commands = pep8 --ignore E501 luigi test examples bin
...
+deps =
+  sqlalchemy
+  Sphinx
+  sphinx_rtd_theme
+
+changedir = doc
+
+commands =
+    make {posargs}
```
Comment (Tarrasch) [inline tox.ini]: How is `{posargs}` different from `{posargs:}`?

**Comment by erikbern** [comment]
Will this trigger build failures if the documentation is broken? That would be pretty cool!

File: tox.ini (hunk 2 above)
Comment (steenzout) [inline tox.ini]: @Tarrasch it seems `{posargs: ...}` is used when you want to set defaults to positional arguments.

http://tox.readthedocs.org/en/1.0/example/general.html

**Comment by steenzout** [comment]
@Tarrasch I've rebased my changes.

**Comment by erikbern** [comment]
I think something went wrong when you rebased – mind trying again?

**Comment by erikbern** [comment]
Also "fixed URL" might not be the best commit msg :)

**Comment by erikbern** [comment]
other than that, LGTM

**Comment by steenzout** [comment]
I'v tweaked the command to generate the docs since it was causing tox to throw a warning.
I've fixed a typo.
I've changed the first line of the commit message.

**Comment by steenzout** [comment]
@erikbern it may be possible to have the build break if the docs generation fails.
I've tried to introduce some errors and make it fail but sphinx is only giving me warnings.
if I find a way I'll send another PR.
will that work?

**Comment by steenzout** [comment]
@erikbern I thought you wanted me to rebase to squash the number of commits.
what do you mean by rebase, exactly?

**Comment by steenzout** [comment]
I think I got it.
pull changes from luigi/master and merge the conflicts?

**Comment by erikbern** [comment]
Yeah "git pull --rebase" or "git rebase master"

**Comment by steenzout** [comment]
got it.

**Comment by steenzout** [comment]
travis failed build due to 'pep8' not being present.
let me have a look...

**Comment by steenzout** [comment]
I figured out that you guys run `autopep8` to fix PEP8 warnings.
I've tweaked it a bit so it doesn't add a new line after class declaration.
I think it doesn't makes sense to have the blank line if the next line is the start of the docstring so I kept in on the `pep8` check (which understands what is going on) but removed it from the `autopep8`.

I figure out how to make build break due to sphinx errors.
I'm preparing another PR for that.

I think the rebase is good.
let me know.

**Comment by erikbern** [comment]
LGTM let's merge

**Comment by Tarrasch** [comment]
Thank you very much! :)

**Pull request merged**